pyccel/ast/core.py

```diff
@@ -5589,28 +5597,29 @@ def is_None(expr):
 
 # ... TODO treat other statements
 
-def get_assigned_symbols(expr):
+def get_assigned_symbols(expr,all):
     """Returns all assigned symbols (as sympy Symbol) in the AST.
 
     Parameters
     ----------
     expr: Expression
         any AST valid expression
+    all: get all symbols if TRUE (ignore rank)
```

Comment:
I don't understand what this does. The docstring should say which type to pass. Is `all` a `bool`?
What does the `rank` have to do with anything? What happens if I pass `false`? It returns no symbols? Only some symbols?

Reply:
yes its a `bool`,
i have added the intent(in) to argument which have `rank = 0`, so when using `get_assigned_symbols() `it get only variable with `rank != 0`, which lead to compilation error for this example:

```
#$ header function funct_c(const int, int)
def funct_c( x, a ):
    x += a
    return x
```
in this case we need to raise an error because the user defined x as a const by also he is modifying it.

Reply:
So if `all == True` then it returns all symbols to which something is assigned? And if `all == False` then it only returns symbols to which something is assigned with `rank!=0`?

Reply:
exactly

Reply:
Ok. Can you update the docstring to make this clearer please. As a minimum you need to provide the expected type. It would be good if you could also specify in the docstring what the expected result is if `all = False`